src/org/scratchOrgErrorCodes.ts

```diff
@@ -38,18 +40,46 @@ const optionalErrorCodeMessage = (errorCode: string, args: string[]): string | u
   }
 };
 
-export const checkScratchOrgInfoForErrors = (
+export const validateScratchOrgInfoForResume = async ({
+  jobId,
+  soi,
+  cache,
+  hubUsername,
+}: {
+  jobId: string;
+  soi: ScratchOrgInfo;
+  cache: ScratchOrgCache;
+  hubUsername: string;
+}): Promise<ScratchOrgInfo> => {
+  if (!soi || !soi.Id || soi.Status === 'Deleted') {
+    // 1. scratch org info does not exist in that dev hub or has been deleted
+    cache.unset(jobId);
+    await cache.write();
```

Comment:
``` suggestion
    await cache.unset(jobId);
```
unset is async and it is also writing the cache.

Reply:
this is the instance method inherited from configStore.  There is a static async method but there was already an instance of the cache floating around so I passed it in instead of readig the file again